src/Xamarin.Android.Build.Tasks/Tests/Xamarin.ProjectTools/Android/AndroidSdkResolver.cs

```diff
@@ -60,6 +60,17 @@ public static string GetJavaSdkPath ()
 			return JavaSdkPath;
 		}
 
+		// Cache the result, so we don't run MSBuild on every call
+		static string DotNetPath;
+		public static string GetDotNetPath ()
+		{
+			if (string.IsNullOrEmpty (DotNetPath))
+				DotNetPath = Environment.GetEnvironmentVariable ("DOTNET_ROOT");
```

Comment:
I was thinking about this, maybe we should take these two lines out?

I don't know of a situation where someone would want their `$DOTNET_ROOT` to take precedence of what is set in `Configuration.props`.

They can use `Configuration.Override.props` to change the location of `$(DotNetPath)`.

Reply:
Maybe we should just change the ordering? Though, I would also be happy to remove this to make test execution more explicit. Maybe it's better to fail if the .NET installation specified in `Configuration.props` can't be found.

Reply:
Reordering like we do for JavaSdkDirectory would be fine:

https://github.com/xamarin/xamarin-android/blob/eb4b49aab01df7a20f78e1c90d03b983e6f17a14/src/Xamarin.Android.Build.Tasks/Tests/Xamarin.ProjectTools/Android/AndroidSdkResolver.cs#L48-L61

Reply:
I removed this, we'll try to not set _any_ variables to point at this preview SDK.